packages/x-data-grid/src/utils/css/themeManager.tsx

```diff
@@ -0,0 +1,64 @@
+import * as React from 'react';
+import { useTheme, type Theme } from '@mui/material/styles';
+import { stylesToString } from './stylesToString';
+import { transformMaterialUITheme } from './transformVariables';
+
+const STYLE_ID = 'mui-x-data-grid';
+const CLASSNAME_PREFIX = 'MuiDataGridVariables';
+
+let element = undefined as HTMLStyleElement | undefined;
+
+let nextClassNameId = 1;
+const classNameByTheme = new WeakMap<object, string>();
+
+export function useCSSVariablesClass() {
+  return getClassNameForTheme(useTheme());
+}
+
+export function GridPortalWrapper({ children }: { children: React.ReactNode }) {
+  const className = useCSSVariablesClass();
+  return <div className={className}>{children}</div>;
+}
+
+function getClassNameForTheme(theme: Theme) {
+  let className = classNameByTheme.get(theme);
+  if (className) {
+    return className;
+  }
+
+  className = `${CLASSNAME_PREFIX}-${nextClassNameId}`;
+  nextClassNameId += 1;
```

Comment:
I'm probably missing something, this can't be stable with hydration, right? There are no guarantees that the order in which the server and client renders is the same with suspense.

Reply:
I'll have to hash the theme then, though it's a one-time operation so it shouldn't be too expensive.